doc/manual_experimental.rst

```diff
@@ -1731,324 +1585,304 @@ e.g. with given example `echo("ab")` will be rewritten just once:
+IO intensive tasks the `async` and `await` features should be
+used instead. Both parallel and spawn need the `threadpool <threadpool.html>`_
+module to work.
 
-A view type is a type that is or contains one of the following types:
+Somewhat confusingly, `spawn` is also used in the `parallel` statement
+with slightly different semantics. `spawn` always takes a call expression of
+the form `f(a, ...)`. Let `T` be `f`'s return type. If `T` is `void`
+then `spawn`'s return type is also `void` otherwise it is `FlowVar[T]`.
 
-- `lent T` (view into `T`)
-- `openArray[T]` (pair of (pointer to array of `T`, size))
+Within a `parallel` section sometimes the `FlowVar[T]` is eliminated
+to `T`. This happens when `T` does not contain any GC'ed memory.
+The compiler can ensure the location in `location = spawn f(...)` is not
+read prematurely within a `parallel` section and so there is no need for
+the overhead of an indirection via `FlowVar[T]` to ensure correctness.
 
-For example:
+**Note**: Currently exceptions are not propagated between `spawn`'ed tasks!
 
-.. code-block:: nim
+This feature is likely to be removed in the future as external packages
+can have better solutions. 
 
-  type
-    View1 = openArray[byte]
-    View2 = lent string
-    View3 = Table[openArray[char], int]
 
+Spawn statement
+---------------
 
-Exceptions to this rule are types constructed via `ptr` or `proc`.
-For example, the following types are **not** view types:
+`spawn`:idx: can be used to pass a task to the thread pool:
 
 .. code-block:: nim
 
-  type
-    NotView1 = proc (x: openArray[int])
-    NotView2 = ptr openArray[char]
-    NotView3 = ptr array[4, lent int]
+  import std/threadpool
 
+  proc processLine(line: string) =
+    discard "do some heavy lifting here"
 
-The mutability aspect of a view type is not part of the type but part
-of the locations it's derived from. More on this later.
+  for x in lines("myinput.txt"):
+    spawn processLine(x)
+  sync()
 
-A *view* is a symbol (a let, var, const, etc.) that has a view type.
+For reasons of type safety and implementation simplicity the expression
+that `spawn` takes is restricted:
 
-Since version 1.4 Nim allows view types to be used as local variables.
-This feature needs to be enabled via `{.experimental: "views".}`.
+* It must be a call expression `f(a, ...)`.
+* `f` must be `gcsafe`.
+* `f` must not have the calling convention `closure`.
+* `f`'s parameters may not be of type `var`.
+  This means one has to use raw `ptr`'s for data passing reminding the
+  programmer to be careful.
+* `ref` parameters are deeply copied which is a subtle semantic change and
+  can cause performance problems but ensures memory safety. This deep copy
+  is performed via `system.deepCopy` and so can be overridden.
+* For *safe* data exchange between `f` and the caller a global `TChannel`
+  needs to be used. However, since spawn can return a result, often no further
+  communication is required.
```

Comment:
```suggestion
* `f`'s parameters may not be of type `var`.
  This means one has to use raw `ptr`'s for data passing, reminding the
  programmer to be careful.
* `ref` parameters are deeply copied, which is a subtle semantic change and
  can cause performance problems, but ensures memory safety. This deep copy
  is performed via `system.deepCopy`, so it can be overridden.
* For *safe* data exchange between `f` and the caller, a global `Channel`
  needs to be used. However, since spawn can return a result, often no further
  communication is required.
```